packages/core/src/sfpcommands/source/DeploySourceToOrgImpl.ts

```diff
@@ -44,169 +43,70 @@ export default class DeploySourceToOrgImpl implements DeploymentExecutor {
-                    null,
-                    this.packageLogger
-                );
-            else
-                SFPLogger.log(
-                    `Deployment is in progress....  Unleashing the power of your code!`,
-                    null,
-                    this.packageLogger
-                );
-
-            // Loop till deployment completes to show status
-            let result;
-            while (true) {
-                try {
-                    result = child_process.execSync(
-                        `sfdx force:mdapi:deploy:report --json -i ${deploy_id} -u ${this.target_org}`,
-                        {
-                            cwd: this.project_directory,
-                            stdio: ['pipe', 'pipe', 'ignore'],
-                            ...defaultProcessOptions(),
-                        }
-                    );
-                } catch (err) {
-                    if (this.deployment_options['checkonly'])
-                        SFPLogger.log(`Validation Failed`, LoggerLevel.ERROR, this.packageLogger);
-                    else SFPLogger.log(`Deployment Failed`, LoggerLevel.ERROR, this.packageLogger);
-                    break;
-                }
-
-                let resultAsJSON = JSON.parse(result);
-                if (resultAsJSON['status'] == 1) {
-                    SFPLogger.log('Deployment Failed', LoggerLevel.ERROR, this.packageLogger);
-                    commandExecStatus = false;
-                    break;
-                } else if (
-                    resultAsJSON['result']['status'] == 'InProgress' ||
-                    resultAsJSON['result']['status'] == 'Pending'
-                ) {
-                    SFPLogger.log(
-                        `${COLOR_TRACE(
-                            `Processing ${resultAsJSON.result.numberComponentsDeployed} out of ${resultAsJSON.result.numberComponentsTotal}`
-                        )}`,
-                        LoggerLevel.INFO,
-                        this.packageLogger
-                    );
-                } else if (resultAsJSON['result']['status'] == 'Succeeded') {
-                    SFPLogger.log(COLOR_SUCCESS('Deployment Succeeded'), LoggerLevel.INFO, this.packageLogger);
-                    commandExecStatus = true;
-                    break;
-                }
-
-                await delay(30000);
-            }
-
-            deploySourceResult.message = await this.getFinalDeploymentStatus(deploy_id);
-            deploySourceResult.result = commandExecStatus;
-            deploySourceResult.deploy_id = deploy_id;
             return deploySourceResult;
         }
     }
 
-    private async getFinalDeploymentStatus(deploy_id: string): Promise<string> {
+    private async displayErrors(result: DeployResult): Promise<string> {
         SFPLogger.log(`Gathering Final Deployment Status`, null, this.packageLogger);
-        let reportAsJSON = '';
-        let deploymentReports = `.sfpowerscripts/mdapiDeployReports`;
```

Comment:
The impact and dependency analysis are dependent on the deployment reports for the component ID, fullName and type.

Reply:
Should be able to write `DeployDetails` to file and maintain backwards compatibility